dataserver.js

```diff
@@ -8,7 +8,7 @@ if (config.backends.data === 'file' ||
     (config.backends.data === 'multiple' &&
      config.backends.metadata !== 'scality') &&
      (config.backends.auth !== 'scality' &&
-      config.backends.metadata !== 'mongodb')) {
```

Comment:
@ssalaues is there any reason why we did check for `config.backends.metadata !== 'mongodb'`

Reply:
The ```(config.backends.auth !== 'scality' && config.backends.metadata !== 'mongodb')``` condition allows for the case where both MongoDB metadata and Scality data backend is enabled and therefore would not require dataserver to be started. 

The issue that caused this addition:
S3 in federation with MongoDB as metadata will fail because there is no Dataserver parameters in the federation setup and shouldn't need Dataserver to be started anyway because of the "Scality" databackend aka RING.


Please be sure to test this on a federation deployment before merging as it may cause issues with federation installs.

Reply:
S3 in federation has config.backends.auth set to 'scality', no?